src/database/models.py

```diff
@@ -4022,6 +4172,153 @@ def full_json(self):
+    Attributes
+    ----------
+    name : str
+      name of the language.
+    code : str
+
+    """
+
+    code = models.CharField(max_length=LANG_CODE_STR_LEN, unique=True)
+    name = models.CharField(max_length=SHORT_STR_LEN, unique=True)
+    is_right_to_left = models.BooleanField(default=False, blank=True) # not used now but maybe used in the future
+
+    def __str__(self):
+        return self.name
+
+
+    def simple_json(self):
+        return model_to_dict(self)
+
+    def full_json(self):
+        return self.simple_json()
+
+    class Meta:
+        db_table = "supported_languages"
+        ordering = ("name",)
+
+
+class CommunitySupportedLanguage(BaseModel):
+    """
+    A class used to represent the languages supported by the platform
+
+    Attributes
+    ----------
+    community : int Foreign key to the community
+    language : int Foreign key to the supported language
+    """
+
+    community = models.ForeignKey(Community, on_delete=models.CASCADE, db_index=True)
+    language = models.ForeignKey(SupportedLanguage, on_delete=models.CASCADE, db_index=True)
+
+    def __str__(self):
+        return f"{self.community.name} - {self.language.name}"
+
+    def simple_json(self):
+        return model_to_dict(self)
+
+    def full_json(self):
+        return self.simple_json()
+
+    class Meta:
+        db_table = "community_supported_languages"
+        unique_together = ["community", "language"]
+        ordering = ("community", "language")
+
+
+class TranslationsCache(RootModel):
+    """
+    A class used to represent the translations cache table
+
+    Attributes
+    ----------
+    hash	: str
+    source_language_code  : str
+    target_language_code  : str
+    translated_text	: str
+    last_translated	: DateTime
+    """
+
+    hash = models.CharField(primary_key=True, max_length=SHORT_STR_LEN, unique=True)
+    source_language_code = models.CharField(max_length=LANG_CODE_STR_LEN)
+    target_language_code = models.CharField(max_length=LANG_CODE_STR_LEN)
+    translated_text = models.TextField(max_length=LONG_STR_LEN)
+    last_translated = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return f"{self.hash}: {self.translated_text}"
+
+    def simple_json(self):
+        return model_to_dict(self)
+
+    def full_json(self):
+        return self.simple_json()
+
+    class Meta:
+        db_table = "translations_cache"
+
+
+class ManualCommunityTranslation(TranslationsCache):
+    """
+    A class used to represent the manual translations done by the community
+
+    Attributes
+    ----------
+    community : int
+    """
+
+    community = models.ForeignKey(Community, on_delete=models.CASCADE, db_index=True)
+
+    def __str__(self):
+        return self.hash
+
+    def simple_json(self):
+        return model_to_dict(self)
+
+    def full_json(self):
+        return self.simple_json()
+
+    class Meta:
+        db_table = "manual_community_translations"
+
+
+class StaticSiteText(BaseModel):
```

Comment:
Delete this, (and the migration file) and re-run the migration to generate a new one and lets merge this.